cfht2caom2/espadons_energy_augmentation.py

```diff
@@ -129,12 +130,19 @@ def _do_energy(artifact, science_fqn, cfht_name):
 
     # read in the complete fits file, including the data
     logging.info(f'Reading ESPaDOnS energy data from {science_fqn}.')
-    hdus = ac.read_fits_data(science_fqn)
-    wave = hdus[0].data[0, :]
+    hdus = fits.open(science_fqn, memmap=True, lazy_load_hdus=False)
+    if hdus[0].data is not None:
+        wave = hdus[0].data[0, :].copy()
+    else:
+        wave = hdus[1].data[0, :].copy()
+    hdr = hdus[0].header.copy()
+    hdus.close()
+    del hdus[0].data
```

Comment:
You've missed the `hdus[1].data` case. I would move these `del`s in the `if/else`